data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/HasTagsExpressionFunction.java

```diff
@@ -21,11 +21,14 @@ public Object evaluate(final List<Object> args, Event event, Function<Object, Ob
         if (args.size() == 0) {
             throw new RuntimeException("hasTags() takes at least one argument");
         }
-        if(!args.stream().allMatch(a -> a instanceof String)) {
-            throw new RuntimeException("hasTags() takes only String type arguments");
+        if(!args.stream().allMatch(a -> ((a instanceof String) && (((String)a).length() > 0) && (((String)a).charAt(0) == '"')))) {
+            throw new RuntimeException("hasTags() takes only non-empty string literal type arguments");
         }
         final List<String> tags = args.stream()
-                                    .map(a -> (String) a)
+                                    .map(a -> {
```

Comment:
You can split map functions so that you can use the compact lambda form. This is not critical though.

```
.map(a -> (String)a)
.map(str -> str.substring(1, str.length() - 1))
```